File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.cc
#include <iostream>
#include <stack>
#define MAX 1001
using namespace std;
int N, M;
int graph[MAX][MAX] = {0,};
bool visited[MAX] = {false,};

void DFS(int i){
    stack<int> s;
    s.push(i);
    while (!s.empty()){
        int x = s.top(); s.pop();
        for (int n = 1; n <= N; n++){
            if (graph[x][n] == 1){
                if (!visited[n]){
                    visited[n] = true;
                    s.push(n);
                }
            }
        }
    }
}

int main(){
    int x, y, cnt = 0;
    cin >> N >> M;
    for (int i = 1; i <= M; i++){
        cin >> x >> y;
        graph[x][y] = 1;
        graph[y][x] = 1;
    }


    for (int i = 1; i <= N; i++){
        if (!visited[i]){
            DFS(i);
            cnt++;
        }
    }

    cout << cnt;

    return 0;
}

File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())
graph = [[i] for i in range(N + 1)]
visited = [0 for _ in range(N + 1)]

for _ in range(M):
    i, j = map(int, input().split())
    graph[i].append(j)
    graph[j].append(i)

def DFS(i):
    stack = [i]
    visited[i] = 1
    while stack:
        x = stack.pop()
        for y in graph[x]:
            if visited[y] == 0:
                visited[y] = 1
                stack.append(y)

cnt = 0
for i in range(1, N+1):
    if visited[i] == 0:
        DFS(i)
        cnt += 1

print(cnt)